packages/hardhat-core/src/internal/artifacts.ts

```diff
@@ -78,7 +78,18 @@ export class Artifacts implements IArtifacts {
   }
 
   public async artifactExists(name: string): Promise<boolean> {
-    const artifactPath = await this._getArtifactPath(name);
+    let artifactPath = "";
```

Comment:
This is not necessary, and it could hide problems that can be caught by the compiler.

```suggestion
    let artifactPath;
```

To understand why this is the case, try commenting out the `throw e` line. If `artifactPath` is not pre-assigned, you'll (correctly) get a compilation error. But if you pre-assign it, then the code will compile, and the runtime behavior will be to call `pathExists` with an empty string.